staging/src/k8s.io/apimachinery/third_party/forked/golang/json/fields.go

```diff
@@ -295,16 +295,19 @@ func dominantField(fields []field) (field, bool) {
 	return fields[0], true
 }
 
-var fieldCache struct {
+type fieldCache struct {
 	sync.RWMutex
 	m map[reflect.Type][]field
 }
+var fldCache = &fieldCache{
+	m: make(map[reflect.Type][]field),
+}
```

Comment:
LGTM

alternatively the struct can be anonymous if the type alias is not used elsewhere:
```
var fieldCache = struct {
	sync.RWMutex
	m map[reflect.Type][]field
}{m: make(map[reflect.Type][]field)}
```

Reply:
Got compilation issue for the above formation on the left curly (preceding 'm:')
```
} {
```

Reply:
probably that the lock is passed by value?

this works:
https://play.golang.org/p/6Up8JsvKDoL

Reply:
Thanks, @neolit123 
That is simpler